app/src/main/java/org/odk/odknotifications/Activities/MainActivity.java

```diff
@@ -351,39 +368,21 @@ public boolean onOptionsItemSelected(MenuItem item) {
     }
 
     private void syncCloudDatabase() {
-        groupArrayList = getGroups();
-        new UnsubscribeNotificationGroups(this).execute();
-        addGroupsFromFirebase();
-        joinODKGroups(groupArrayList);
-        addMenuItemInNavMenuDrawer();
-    }
 
-    private void addGroupsFromFirebase() {
-        final ProgressDialog dialog = new ProgressDialog(this);
-        dialog.setMessage("Please Wait! Loading groups from firebase...");
-        dialog.show();
-        dialog.setCancelable(false);
-        DatabaseReference mRef  = FirebaseDatabase.getInstance().getReference().child("clients").child(loggedInUsername).child("groups");
+        try {
+            groupArrayList = ServerDatabaseCommunicator.getInstance().getGroupsList(getActiveUser());
+            notificationArrayList = ServerDatabaseCommunicator.getInstance().getNotifications();
```

Comment:
`When ServerDatabaseCommunicator.getInstance().getNotifications()` is called it changes the reference of `notificationArrayList` which is going to be different from the one you passed in `NotificationAdapter` so when you call `notificationAdapter.notifyDataSetChanged()` the adapter or the list is not going to be updated as it still have the old reference.

So to solve this problem you can do one thing. 
Step 1. Call `notificationArrayList.clear()` it will basically clear all your previous notification and avoid duplication of notification because `getNotifications()` will anyways fetch all of them again.
Step 2. `notificationArrayList.addAll(ServerDatabaseCommunicator.getInstance().getNotifications())`
This will bring all the notification from the database and add that to the same list that you passed to a `NotificationAdapter`

Reply:
Done. Thanks a lot for pointing out this error. Completely missed this one.